### PR TITLE
READY: Improve erlang:now deprecation define regex

### LINKED ISSUE #284: platform define for deprecated_now is limited to OTP 18 [JIRA: CLIENTS-857]
Because it is only set for 18 but is also required for all versions after 18 the build fails on 19: https://github.com/basho/riak-erlang-client/blob/master/rebar.config#L11

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes #284 (CLIENTS-51) (CLIENTS-525) (CLIENTS-857)

### CODE REVIEW

**Comment by javajolt** [comment]
As previously discussed, this crashes on OTP 19rc1:
```
./rebar get-deps
{"init terminating in do_boot",{load_failed,[ets,supervisor,kernel,gen_server,file_io_server,file,erl_parse,erl_lint,erl_eval,code,application_master,application,application_controller,proc_lib,gen_event,lists,error_handler,heart,gen,file_server,filename,error_logger,code_server]}}

Crash dump is being written to: erl_crash.dump...done
init terminating in do_boot ()
make: *** [deps] Error 1
```
Retesting with master

**Comment by javajolt** [comment]
Passes with OTP 19 master

File: README.md [hunk 1]
```diff
@@ -673,10 +635,23 @@ SecurityOptions = [
...
+More detailed information can be found in our [official documentation][kv_using_security_basics].
 
 Troubleshooting
 ==================
 
 If `start/2` or `start_link/2` return `{error,econnrefused}` the client could not connect to the server - make sure the protocol buffers interface is enabled on the server and the address/port is correct.
+
+[basho_docs]: http://docs.basho.com/
+[kv_setup]: http://docs.basho.com/riak/kv/latest/setup/
+[erlang_client_edocs]: http://basho.github.com/riak-erlang-client/
```
Comment (javajolt) [inline README.md]: These docs appear to be ancient, i.e., 2013.  Should we either update or remove the link?

Comment (lukebakken) [inline README.md]: Yeah, we should update them. I'll look into how to do that.

**Comment by javajolt** [comment]
:+1: 23e8d5b

**Comment by javajolt** [comment]
@borshop merge

**Pull request merged**